app/src/androidTest/java/com/vero/tasky/auth/presentation/LoginE2E.kt

```diff
@@ -104,12 +74,42 @@ class LoginE2E {
                             }
                         },
                     ) { paddingValues ->
-                        RootNavigationTest(
-                            navController = rememberNavController(),
-                            isLoggedIn = composeRule.activity.viewModel.state.collectAsState().value.isLoggedIn,
-                            modifier = Modifier.padding(paddingValues),
-                            loginViewModel = loginViewModel
-                        )
+                        val navController = rememberNavController()
+                        NavHost(
+                            navController = navController,
+                            startDestination = if (
+                                composeRule.activity.viewModel.state.collectAsState().value.isLoggedIn
+                            ) Screens.Agenda.route else Screens.Login.route,
+                            modifier = Modifier.padding(paddingValues)
+                        ) {
+                            composable(route = Screens.Login.route) {
+                                LoginScreen(
+                                    onSignUp = { navController.navigate(Screens.Registration.route) },
+                                )
+                            }
+                            composable(route = Screens.Agenda.route) {
+                                AgendaScreen(
+                                    onAgendaItemClick = { agendaItem, isEditable ->
+                                        val route = when (agendaItem) {
+                                            is AgendaItem.Event -> Screens.Event.route + "?${NavigationConstants.EVENT_ITEM_ID}"
+                                            is AgendaItem.Task -> Screens.Task.route + "?${NavigationConstants.TASK_ITEM_ID}"
+                                            is AgendaItem.Reminder -> Screens.Reminder.route + "?${NavigationConstants.REMINDER_ITEM_ID}"
+                                        }
+                                        val parameters =
+                                            "=${agendaItem.id}&${NavigationConstants.IS_EDITABLE}=$isEditable"
+                                        navController.navigate("$route$parameters")
+                                    },
+                                    onNewAgendaItemClick = { agendaItemType ->
+                                        val route = when (agendaItemType) {
+                                            AgendaItemType.REMINDER -> Screens.Reminder.route
+                                            AgendaItemType.TASK -> Screens.Task.route
+                                            else -> Screens.Event.route
+                                        }
+                                        navController.navigate(route)
+                                    }
+                                )
+                            }
```

Comment:
You still implement a copy of the production logic, which is a no go for tests. If your production logic changes, the test logic will remain and keep on passing, even if there's a bug in the production code.
Instead, just include your root composable here which contains the whole nav host:
```
// Production code
@Composable
fun AppRoot(navController: NavHostController) {
    // Your navigation logic here
}

------------------

// Test code
composeRule.setContent {
    val navController = rememberNavController()
    AppRoot(navController)
}
```
You can also take a look at `TestNavController` for more control.